jdbc/src/main/spotbugs/spotbugs-exclude.xml

```diff
@@ -18,4 +18,9 @@
     <Class name="software.amazon.timestream.jdbc.TimestreamTablesResultSet"/>
     <Method name="populateCurrentRows"/>
   </Match>
+  <Match>
+    <Bug pattern="SQL_NONCONSTANT_STRING_PASSED_TO_EXECUTE"/>
```

Comment:
Do we have options to fix it, so we don't need to exclude it?

Reply:
The cause of spotbugs reporting this as a bug is because `"SHOW DATABASES LIKE 'pattern'"` string is dynamically constructed (when a pattern is provided). I just spent some time trying to use a prepared statement, but the function stopped working as a result because statement.setString(int, String) is [not supported, click here for function header code](https://github.com/Bit-Quill/amazon-timestream-driver-jdbc/blob/main/jdbc/src/main/java/software/amazon/timestream/jdbc/TimestreamPreparedStatement.java#L403).

To fix this `populateCurrentRows` function, `statement.setString(int, String)` needs to be implemented first. 

-------------------------------------------------------------------
The code that I tried for `populateCurrentRows(TimestreamConnection connection, String schemaPatter)`:
It solves the spotbugs error but the driver does not load anything for tables
```
if (Strings.isNullOrEmpty(schemaPattern)){
      try (PreparedStatement statement = connection.prepareStatement("SHOW DATABASES")) {
        try (ResultSet rs = statement.executeQuery()) {
          while (rs.next()) {
            databases.add(new Row().withData(
                    new Datum().withScalarValue(rs.getString(1)),
                    NULL_DATUM
            ));
          }
        }
      }
    } else {
      try (PreparedStatement statement = connection.prepareStatement("SHOW DATABASES LIKE '?'")) {
        statement.setString(1,schemaPattern);
        try (ResultSet rs = statement.executeQuery()) {
          while (rs.next()) {
            databases.add(new Row().withData(
                    new Datum().withScalarValue(rs.getString(1)),
                    NULL_DATUM
            ));
          }
        }
      }
    }
```

Reply:
What can be done not to include this exclude?

Reply:
To recap, this exclude is for function `TimestreamSchemasResultSet`. To not include this exclude, I need to implement the function `TimestreamSchemasResultSet` to use prepared statements, which requires prepared statements to be fully implemented first. The implementation is non-trivial and is out of scope.

Reply:
There was already an exclude for `populateCurrentRows` for `TimestreamTablesResultSet` in this file. This is due to the same reason: lack of support for prepared statements.